### DISCUSSION TITLE
Add Radio Garden to the great list

### DISCUSSION THREAD

**Discussion by afkcodes** [junguler/m3u-radio-music-playlists#33] [General]
Hey thanks for this amazing repo, loved it. I have two Questions
1. Can we add [radio Garden](http://radio.garden/) to the repo
2. are the channels are checked for playablity status

**Comment by junguler** [comment]
hello, thanks for the kind words

1. radio garden is on my todo list, the thing is it's set up in a weird way, to be honest i'm not sure how i need to go about scraping it, there is one [github repo](https://github.com/mcplayer9999/radio-garden-m3u) that has it scraped already but it's kind of old
2. these are not checked as i'm only scraping the data from the websites, i have [a script](https://github.com/junguler/m3u-radio-music-playlists/blob/main/stuff/check-for-dead.sh) that checks them for status code 200 and removes everything else but it would take a long time for me to go thru everything, if what you want is on a small scale it's best to just run it yourself locally or on github actions, otherwise i may need to look for a more efficient and faster way to remove offair or broken streams in the future

**Comment by afkcodes** [comment]
i do have a python script that i use to check streams i can help you with the same. it uses vlc instance to check the streams but, i have to use vpn for efficient checking for different countries. let me know if i can help in any way

**Comment by junguler** [comment]
please link your script so i can check and compare it with my own, my internet speed is awful so i need to use it on github actions, similar to how i scrape websites

speed is the biggest factor here because i have more than 500k unique stream links to check from, not to mention all of the duplicates each folder inevitably going to have, this is a big undertaking regardless tho and even if i were to go thru everything it would take quite a while to finish

i might need to strip each file from the +merged+ folder to a 100 streams or so and create a +checked+ folder of some kind but those will be streams that are compatible with azure ips since they are going to be checked with github actions

it's complicated and time consuming but i'm not against the idea at all

**Comment by afkcodes** [comment]
sure i have not pushed it yet will push it and link it here

**Comment by junguler** [comment]
i see, i'm not in a rush, i'll convert this to a discussion if it's fine by you, we can keep the conversation going there

**Reply by afkcodes** [reply]
I have a better bandwidth 300 Mbps I can check the streams at my side, but i will be needing the stations de-duplicated and segregated in smaller files better if we have it categorised in according to countries

**Reply by junguler** [reply]
i already have many stream files sorted by countries, you can do a little test on the [live_online-raido](https://github.com/junguler/m3u-radio-music-playlists/tree/main/live_online_radio) folder for example, like i said it's too much effort to go thru and do everything and i'm not even sure if it's going to be worth it, i think i'll let people do this themselves because every country and area is going to have different results due to geo restricted content

**Reply by afkcodes** [reply]
agree to this. i will test it out and let you know. I know it will be a time consuming task to check through all the streams and as you mentioned the geo restriction is one factor

**Comment by junguler** [comment]
hello again, i figured since this discussion with you prompted me to try again with checking links and i was able to fix the issue of it being too slow i would comment here to make you aware of the newly made [+checked+](https://github.com/junguler/m3u-radio-music-playlists/tree/main/%2Bchecked%2B) folder, it's checked against azure servers using github actions, i'm 90% sure it's based on america so keep the geo restricted content in mind

**Reply by afkcodes** [reply]
surely gonna check them out, thanks

**Comment by junguler** [comment]
hello again, i've finally figured out how to scrape radio.garden, you can find it [here](https://github.com/junguler/m3u-radio-music-playlists/tree/main/radio.garden), hope you like it

**Comment by afkcodes** [comment]
Awesome, Thanks for the info
